### ISSUE TITLE
What did you think about the process of writing a command?

### ISSUE THREAD

**Issue by danielbachhuber** [itspriddle/wp-cli-tgmpa-plugin#12]
What went well, and what could be improved?

**Comment by itspriddle** [comment]
tl;dr: WP-CLI's well documented/organized code made it pretty easy, but I struggled a bit with the test suite because I wasn't familiar with Behat and haven't worked much with Cucumber style tests.

---

Writing the actual command was pretty straightforward. It is mostly a wrapper around the existing `wp plugin` subcommands. Implementing those was just just a matter of grokking WP-CLI's `php/commands/plugin.php` to see how things were implemented. Where I needed custom logic beyond spawning a `wp plugin` command with `WP_CLI::run_command()`, I was glad to see that most of WP-CLI is pretty well documented. That was a huge help. I **really** like how the PHPdoc stuff in commands works to validate arguments. It forces you to write proper documentation and I think that is a great feature.

Implementing the test suite was where I struggled a bit. That was mostly due to my unfamiliarity with Behat and Cucumber style tests. I found that WP-CLI's test suite is pretty comprehensive, and it was easy to poke through your features to figure out what I needed to do. Bonus points for including tests in the scaffold -- I think its great that you're advocating testing!

**Comment by danielbachhuber** [comment]
👍  Thanks! If any documentation improvements cross your mind, I'd be happy to hear about them.
